src/tests/rpp/test_subjects.cpp

```diff
@@ -255,4 +256,160 @@ TEST_CASE("serialized_subject handles race condition")
+
+        size_t bound = 1;
+        auto   sub   = rpp::subjects::replay_subject<int>{bound};
+
+        SECTION("subscribe multiple observers")
+        {
+            sub.get_observable().subscribe(mock_1.get_observer());
+
+            sub.get_observer().on_next(1);
+            sub.get_observer().on_next(2);
+            sub.get_observer().on_next(3);
+
+            SECTION("observer obtains values")
+            {
+                CHECK(mock_1.get_received_values() == std::vector{1, 2, 3});
+                CHECK(mock_1.get_total_on_next_count() == 3);
+                CHECK(mock_1.get_on_error_count() == 0);
+                CHECK(mock_1.get_on_completed_count() == 0);
+            }
+
+            sub.get_observable().subscribe(mock_2.get_observer());
+
+            SECTION("observer obtains latest replayed values")
+            {
+                CHECK(mock_2.get_received_values() == std::vector{3});
+                CHECK(mock_2.get_total_on_next_count() == 1);
+                CHECK(mock_2.get_on_error_count() == 0);
+                CHECK(mock_2.get_on_completed_count() == 0);
+            }
+        }
+    }
+
+    SECTION("bounded replay subject with duration")
+    {
+        using namespace std::chrono_literals;
+
+        auto mock_1 = mock_observer_strategy<int>{};
+        auto mock_2 = mock_observer_strategy<int>{};
+
+        size_t bound    = 2;
+        auto   duration = 5ms;
+        auto   sub      = rpp::subjects::replay_subject<int>{bound, duration};
+
+        SECTION("subscribe multiple observers")
+        {
+            sub.get_observable().subscribe(mock_1.get_observer());
+
+            sub.get_observer().on_next(1);
+            sub.get_observer().on_next(2);
+            sub.get_observer().on_next(3);
+
+            SECTION("observer obtains values")
+            {
+                CHECK(mock_1.get_received_values() == std::vector{1, 2, 3});
+                CHECK(mock_1.get_total_on_next_count() == 3);
+                CHECK(mock_1.get_on_error_count() == 0);
+                CHECK(mock_1.get_on_completed_count() == 0);
+            }
+
+            std::this_thread::sleep_for(duration);
+            sub.get_observer().on_next(4);
+
+            sub.get_observable().subscribe(mock_2.get_observer());
+
+            SECTION("subject replay only non expired values")
+            {
+                CHECK(mock_2.get_received_values() == std::vector<int>{4});
+                CHECK(mock_2.get_total_on_next_count() == 1);
+                CHECK(mock_2.get_on_error_count() == 0);
+                CHECK(mock_2.get_on_completed_count() == 0);
+            }
+        }
+    }
+}
+
+TEST_CASE("replay subject doesn't introduce additional copies")
+{
+    auto sub = rpp::subjects::replay_subject<copy_count_tracker>{};
+
+    SECTION("subscribe")
+    {
+        sub.get_observable().subscribe([](const copy_count_tracker& tracker){
+            CHECK(tracker.get_copy_count(), 1); // 1 copy to internal replay buffer
```

Comment:
@victimsnino why do we enforce const lvalue ref on subjects strategies? In replay case, allowing forwarding ref would prevent this copy

Reply:
This copy is result of copying to buffer. If you would forward value to buffer, it would be 1 move instead, isn't it?
Anyway, there is no enforcing of this thing, only requirement, that subject have to pass emisions as const lvalue to its observers due to we can't forward value to first one if we have multiple observers